doc/1.4/running-dmlc.md

```diff
@@ -115,6 +115,11 @@ causes generated C code to follow the DML code more closely.
 
 Adds Synopsys® Coverity® analysis annotations to suppress common false positives
 in generated C code created from DML 1.4 device models.
+
+Analysis annotation generation impacts the generation of line directives in a
+way that may cause debugging or coverage tools besides Coverity to display
+unexpected behavior. Because of this, it's recommended that `--coverity` is
+only used when needed.
```

Comment:
So this disclaimer specifically refers to this inside `coverity_markers`:
```
out('#ifndef __COVERITY__\n')
site_linemark_nocoverity(site, adjust=-(len(markers) + 1))
out('#endif\n')
```
The linemark we generate is adjusted such that the next line that the upcoming line after the `#endif` and the `/* coverity ... */ markers will be redirected to the correct DML line; however, that means that the `#endif` and the markers will be redirected to the _preceding_ DML lines. This makes no difference if you're only interested in Coverity and GCC error reporting working properly; they will never complain about the `#endif` or comments. However, GDB may perhaps consider breaking on an `#endif` or a comment (however that works out), and GCOV may consider the redirected DML lines covered.

Reply:
And no, there is no way around this save by omitting the `#ifndef` entirely, in which case the preceding line directive reset will still be in effect which would _also_ be unexpected,  and so this disclaimer would still needed save it would also need to say that GCC error reporting may display unexpected behavior.

Reply:
This is why I hate Coverity analysis annotations. You hate them because their implementation necessitates pragmas. I hate them because their implementation necessitates _this._

Reply:
And no, there is no superior option to analysis annotations either. Believe me, I've tried to find one.